File: 02_2017/Java/Hexagon.java
public class Hexagon extends Shape {

    public static double AREA_CONSTANT = 3 * Math.sqrt(3) / 2;

    public double sideLength;

    public Hexagon(double sideLength) {
        this.sideLength = sideLength;
        if (sideLength < 0)
            throw new IllegalArgumentException("Side length cannot be negative");
    }

    @Override
    public double surfaceArea() {
        return AREA_CONSTANT * sideLength * sideLength;
    }

    @Override
    public double perimeter() {
        return 6 * sideLength;
    }

}

File: 02_2017/cpp/Hexagon.cpp
#include "Hexagon.hpp"

#include <cmath>

static const double area_constant = 3 * std::sqrt(3) / 2;

Hexagon::Hexagon(double sideLength) : sideLength {sideLength} {}

double Hexagon::surface_area() noexcept { return area_constant * sideLength * sideLength; }

double Hexagon::perimeter() noexcept { return 6 * sideLength; }